driver-core/src/test/functional/com/mongodb/ClusterFixture.java

```diff
@@ -396,32 +400,41 @@ public static StreamFactory getStreamFactory() {
     }
 
     public static StreamFactory getAsyncStreamFactory() {
-        String streamType = System.getProperty("org.mongodb.test.async.type", "nio2");
-
-        if (streamType.equals("netty")) {
-            return new NettyStreamFactory(getSocketSettings(), getSslSettings());
-        } else if (streamType.equals("nio2")) {
+        StreamFactoryFactory overriddenStreamMetafactory = getOverriddenStreamFactoryFactory();
```

Comment:
Naming convention question: why `Metafactory` and not `FactoryFactory`?

Reply:
Thought I would use "metafactory" as this is what should have been used instead of "factoryFactory". Replaced with "factoryFactory".

Done.